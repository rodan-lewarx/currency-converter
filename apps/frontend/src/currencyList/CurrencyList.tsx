import { Currency } from "@task/currency-list-schema";
import { Table } from "../ui-components";

export const CurrencyList = ({
  currencyList,
}: {
  currencyList: Currency[];
}) => {
  return (
    <Table>
      <thead>
        <tr>
          <th>Country</th>
          <th>Currency</th>
          <th>Amount</th>
          <th>Currency Code</th>
          <th>Rate</th>
        </tr>
      </thead>
      <tbody>
        {currencyList.map((currency) => (
          <tr key={currency.currencyCode}>
            <td>{currency.country}</td>
            <td>{currency.currency}</td>
            <td>{currency.amount}</td>
            <td>{currency.currencyCode}</td>
            <td>{currency.rate}</td>
          </tr>
        ))}
      </tbody>
    </Table>
  );
};
