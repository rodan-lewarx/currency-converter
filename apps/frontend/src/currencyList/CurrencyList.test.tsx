import { describe, expect, it, afterEach } from "vitest";
import { CurrencyList } from "./CurrencyList";
import { render, cleanup } from "@testing-library/react";

/**
 * @vitest-environment jsdom
 */

const currencyList = [
  {
    country: "United Kingdom",
    currency: "pound sterling",
    amount: 1,
    currencyCode: "GBP",
    rate: 28.222,
  },
  {
    country: "United States",
    currency: "US Dollar",
    amount: 1,
    currencyCode: "USD",
    rate: 22.781,
  },
];

afterEach(async () => {
  cleanup(); // clear testing data after each test run
});

describe("CurrencyList", () => {
  it("should render a table", () => {
    const table = render(<CurrencyList currencyList={currencyList} />);
    expect(table).toMatchSnapshot();
  });
});
