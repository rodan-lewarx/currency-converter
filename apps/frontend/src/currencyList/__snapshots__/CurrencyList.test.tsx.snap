// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CurrencyList > should render a table 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="sc-dlniIP bQqbFZ"
      >
        <thead>
          <tr>
            <th>
              Country
            </th>
            <th>
              Currency
            </th>
            <th>
              Amount
            </th>
            <th>
              Currency Code
            </th>
            <th>
              Rate
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              United Kingdom
            </td>
            <td>
              pound sterling
            </td>
            <td>
              1
            </td>
            <td>
              GBP
            </td>
            <td>
              28.222
            </td>
          </tr>
          <tr>
            <td>
              United States
            </td>
            <td>
              US Dollar
            </td>
            <td>
              1
            </td>
            <td>
              USD
            </td>
            <td>
              22.781
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="sc-dlniIP bQqbFZ"
    >
      <thead>
        <tr>
          <th>
            Country
          </th>
          <th>
            Currency
          </th>
          <th>
            Amount
          </th>
          <th>
            Currency Code
          </th>
          <th>
            Rate
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            United Kingdom
          </td>
          <td>
            pound sterling
          </td>
          <td>
            1
          </td>
          <td>
            GBP
          </td>
          <td>
            28.222
          </td>
        </tr>
        <tr>
          <td>
            United States
          </td>
          <td>
            US Dollar
          </td>
          <td>
            1
          </td>
          <td>
            USD
          </td>
          <td>
            22.781
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
